src/libs/actions/BankAccounts.js

```diff
@@ -1,11 +1,128 @@
+        });
+}
+
+/**
+ * List of bank accounts. This data should not be stored in Onyx since it contains unmasked PANs.
+ *
+ * @private
+ */
+let plaidBankAccounts = [];
+let bankName = '';
+let plaidAccessToken = '';
+
+/**
+ * @param {String} publicToken
+ * @param {String} bank
+ */
+function getPlaidBankAccounts(publicToken, bank) {
+    bankName = bank;
+
+    Onyx.merge(ONYXKEYS.PLAID_BANK_ACCOUNTS, {loading: true});
+    API.BankAccount_Get({
+        publicToken,
+        allowDebit: false,
+        bank,
+    })
+        .then((response) => {
+            plaidAccessToken = response.plaidAccessToken;
+            plaidBankAccounts = response.accounts;
+            Onyx.merge(ONYXKEYS.PLAID_BANK_ACCOUNTS, {
+                error: {
+                    title: response.title,
+                    message: response.message,
+                },
+                loading: false,
+                accounts: _.map(plaidBankAccounts, account => ({
+                    ...account,
+                    accountNumber: Str.maskPAN(account.accountNumber),
+                })),
+            });
+        });
+}
+
+/**
+ * We clear these out of storage once we are done with them so the user must re-enter Plaid credentials upon returning.
+ */
+function clearPlaidBankAccounts() {
+    plaidBankAccounts = [];
+    bankName = '';
+    Onyx.set(ONYXKEYS.PLAID_BANK_ACCOUNTS, {});
+}
+
+/**
+ * Adds a bank account via Plaid
+ *
+ * @param {Object} account
+ * @param {String} password
+ * @param {String} plaidLinkToken
+ */
+function addPlaidBankAccount(account, password, plaidLinkToken) {
+    const unmaskedAccount = _.find(plaidBankAccounts, bankAccount => (
+        bankAccount.plaidAccountID === account.plaidAccountID
+    ));
+    API.BankAccount_Create({
+        accountNumber: unmaskedAccount.accountNumber,
+        addressName: unmaskedAccount.addressName,
+        allowDebit: false,
+        confirm: false,
+        isSavings: unmaskedAccount.isSavings,
+        password,
+        routingNumber: unmaskedAccount.routingNumber,
+        setupType: 'plaid',
+        additionalData: JSON.stringify({
+            useOnFido: false,
+            policyID: '',
+            plaidLinkToken,
+            isInSetup: true,
+            bankAccountInReview: null,
+            currentStep: 'AccountOwnerInformationStep',
+            bankName,
+            plaidAccountID: unmaskedAccount.plaidAccountID,
+            ownershipType: '',
+            acceptTerms: true,
+            country: 'US',
+            currency: 'USD',
+            fieldsType: 'local',
+            plaidAccessToken,
+            isForStripeConnect: false,
+        }),
+    })
+        .then((response) => {
+            if (response.jsonCode !== 200) {
+                alert('There was a problem adding this bank account.');
+                return;
+            }
+
+            alert('Bank account added successfully.');
+        });
```

Comment:
Is there a way to "close" the modal at this point, e.g. go back to the last route that was used? I don't know if it's already supposed to, but in my tests the modal stayed open, and I can't see anything that suggests we would close the modal here.

Reply:
We can do it, but mostly am just trying to do very simple things for now. Eventually we might redirect you to the Payments preferences in settings (or something else), but the UI hasn't been built yet.

Reply:
Sounds good!